starknet/pre_confirmed_block.go

```diff
@@ -0,0 +1,33 @@
+package starknet
+
+import "github.com/NethermindEth/juno/core/felt"
+
+// Block object returned by the feeder in JSON format for "get_preconfirmed_block" endpoint
+type PreConfirmedBlock struct {
+	Status           string                `json:"status"`
+	Transactions     []*Transaction        `json:"transactions"`
```

Comment:
Why is this `[]*Transaction` instead of `[]Transaction`? Is it because the transaction is optional, or because of how some library or existing code work?